### ISSUE TITLE
[frameit] Document option `force_device_type`

### ISSUE THREAD

**Issue by janpio** [fastlane/fastlane#14077]
Frameit has a `force_device_type` option that is not really documented.

From the source code, it seems you can set it to a string, that is then used to force the type of frame to be used: 
https://github.com/fastlane/fastlane/blob/551a82fe22bfbadb8e044d95c09b5d47d78f522c/frameit/lib/frameit/options.rb#L30-L39
instead of using the screen size to find out the device as usual:
https://github.com/fastlane/fastlane/blob/dcc95a64edde0bf616a62bcc62f3737f7981d3e9/frameit/lib/frameit/screenshot.rb#L24

To be honest I am not even sure that this option (still) works as I think it does, so this has to be tested first.

The file to be edited: https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/docs/frame_screenshots.md

**Comment by janpio** [comment]
Nice @mollyIV, that was pretty much the conclusion I came to when looking into this a bit - `force_device_type` is just broken. (Worse, even the normal device frame selection is broken, see https://github.com/fastlane/fastlane/pull/14142#issue-247765687 for some examples.)

The question is what we do about this.

I am currently rewriting the frameit internals to split the framing from the wrapping in https://github.com/fastlane/fastlane/issues/14088 - this will probably make it easier to then later understand what is going on here and how to fix it.

(That being said, I removed the "you can do this" issue as this turned out to be much more complicated than I initially thought.)

What do you think @mollyIV?

**Comment by mollyIV** [comment]
I think you are absolutely right @janpio 👍It looks like before doing this ticket, we need to create a new issue about fixing `force_device_type` parameter. Then I guess we can establish the relationship between those two - the docs update is blocked by a bug, right? 

I've noticed you created a few more tickets related to updating the docs for `fameit` tool. Since this issue is on hold, I am going to have a look at another one (`force_orientation_block`) 💪

**Comment by janpio** [comment]
Yeah, making it work before documenting might be a good idea 🙈 

How would you have _expected_ it to work though?

**Comment by mollyIV** [comment]
According to the parameter description in the code:
> "Forces a given device type, useful for Mac screenshots, as their sizes vary"
so it's kind of hard to say what to expect 😅 

I was trying to image how I'd like to use a given parameter. 

**1. Solution** 
Let's say we have screenshots from iPhone XS and iPhone 6 Plus in our `screenshots` catalog. By setting `force_device_type: "iPhone_6_Plus"` I'd expect to get framed screenshots **only** for the screenshots from iPhone 6 Plus. 
If there were not any of iPhone 6 Plus screenshots, it would frame nothing. 

The behaviour might get a bit complicated for Mac though - your screenshots might have different resolutions, so what would you expect by setting `force_device_type: "Mac"`? 🤔 Maybe we should add additional MacBook types, just like for the iPhones? 🤷‍♂️ 

**2. Solution**
Another solution could be forcing all the screenshots in the screenshot catalog to be framed by forced device. Let's say we have screenshots from iPhone XS and iPhone 6 Plus in our `screenshots` catalog. By setting `force_device_type: "iPhone_6_Plus"` I'd expect to get **all** screenshots framed into iPhone 6 Plus frame. 

I'd give a go for solution number 1 👍 

WDYT @janpio , does that make sense? 🤔

**Comment by janpio** [comment]
The code points to solution 2.

A parameter with behavior of solution 1 would probably be named `limit_screenshots_to_device_type` or `only_device_type`.

But maybe this is also an opening to remove this broken feature if nobody is using it and it isn't well understood anyway... Will have to investigate when and how this was added with which reasoning.
